dali/core/nvtx.cc

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/core/nvtx.h"
+
+namespace dali {
+
+#if NVTX_ENABLED
+class DomainTimeRangeImpl {
+ public:
+  DomainTimeRangeImpl() {
+    dali_domain_ = nvtxDomainCreateA("DALI");
+  }
+
+  ~DomainTimeRangeImpl() {
+    nvtxDomainDestroy(dali_domain_);
+  }
+
+  void Start(const std::string name, const uint32_t rgb) {
+    nvtxEventAttributes_t att = {};
+    att.version = NVTX_VERSION;
+    att.size = NVTX_EVENT_ATTRIB_STRUCT_SIZE;
+    att.colorType = NVTX_COLOR_ARGB;
+    att.color = rgb | 0xff000000;
+    att.messageType = NVTX_MESSAGE_TYPE_ASCII;
+    att.message.ascii = name.c_str();
```

Comment:
Maybe deduplicate filling the attributes?
But I don't know if it's worth, maybe until it doesn't need to diverge, we can put it in the base class?

Reply:
Done